corehq/apps/locations/templates/locations/manage/partials/drilldown_location_widget.html

```diff
@@ -15,6 +15,10 @@
             data: {'results': $(e).select2('data')},
         }
         $('#{{ css_id }}').select2(options);
+        option_ids = _.map(options.data.results, function(option) { return option.id })
+        if(!option_ids.includes($('#{{ css_id }}').val())) {
+            $('#{{ css_id }}').val(null);
+        }
```

Comment:
Even though the options for primary location was cleared by 'select2' its value was still retained. Which was later handled at server though. Hence made this change.